pkg/metrics_store/metrics_writer.go

```diff
@@ -101,25 +101,45 @@ func SanitizeHeaders(contentType string, writers MetricsWriterList) MetricsWrite
 
 				// Removes duplicate headers from the given MetricsWriterList for the same family (generated through CRS).
 				// These are expected to be consecutive since G** resolution generates groups of similar metrics with same headers before moving onto the next G** spec in the CRS configuration.
-				if header == lastHeader {
-					writer.stores[0].headers[i] = ""
-				} else if strings.HasPrefix(header, "# HELP") {
-					lastHeader = header
+				// Skip this step if we encounter a repeated header, as it will be removed.
+				if header != lastHeader && strings.HasPrefix(header, "# HELP") {
 
 					// If the requested content type was proto-based, replace "info" and "statesets" with "gauge", as they are not recognized by Prometheus' protobuf machinery.
 					if strings.HasPrefix(contentType, expfmt.ProtoType) {
 						infoTypeString := metric.Info.String()
```

Comment:
looks like you only need those lengths here, it might be easier to store them in a global var? (Not sure if the compiler will optimize this)

Reply:
Ah nvm, you're comparing as well. Should be all fine.